src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -513,7 +521,7 @@ public async Task UnconfigureAsync(CommandSettings command)
                         serviceControlManager.UnconfigureService();
                         _term.WriteLine(StringUtil.Loc("Success") + currentAction);
                     }
-                    else if (PlatformUtil.RunningOnLinux)
+                    else if (PlatformUtil.RunningOnLinux || PlatformUtil.RunningOnFreeBSD)
```

Comment:
FreeBSD doesn't run systemd, this probably needs to be its own branch.

Reply:
Yes, this should be in its own `else if` statement.
Could someone then please also upload a corresponding web page for FreeBSD to which the error message refers, as already for OSX and Linux?
https://github.com/microsoft/azure-pipelines-agent/blob/784ab7c16973e7caf03fc9cd12df67a22f9dc3d6/src/Misc/layoutbin/en-US/strings.json#L612-L613

Reply:
Split exceptions into separate `else if` statements and fixed a typo by renaming `UnconfigureServiceDService` to `UnconfigureSystemDService`.